pkg/discovery/dtofactory/node_group_entity_dto_builder.go

```diff
@@ -0,0 +1,59 @@
+package dtofactory
+
+import (
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/repository"
+	sdkbuilder "github.com/turbonomic/turbo-go-sdk/pkg/builder"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type nodeGroupEntityDTOBuilder struct {
+	clusterSummary *repository.ClusterSummary
+}
+
+const (
+	affinityCommodityDefaultCapacity = 1e10
+)
+
+func NewNodeGroupEntityDTOBuilder(clusterSummary *repository.ClusterSummary) *nodeGroupEntityDTOBuilder {
+	return &nodeGroupEntityDTOBuilder{
+		clusterSummary: clusterSummary,
+	}
+}
+
+// Build entityDTOs based on the given node list.
+func (builder *nodeGroupEntityDTOBuilder) BuildEntityDTOs() ([]*proto.EntityDTO, error) {
+	var result []*proto.EntityDTO
+	nodeGrp2nodes := make(map[string]sets.String) // Map of nodeGroup ---> nodes
+	for _, node := range builder.clusterSummary.Nodes {
+		for key, value := range node.ObjectMeta.Labels {
+			fullLabel := key + "=" + value
+			nodeLst := nodeGrp2nodes[fullLabel]
+			if nodeLst.Len() == 0 {
+				nodeLst = sets.NewString(string(node.UID))
+			} else {
+				nodeLst.Insert(string(node.UID))
+			}
+			nodeGrp2nodes[fullLabel] = nodeLst
+		}
+	}
+	// start building the NodeGroup entity dto
+	for fullLabel, _ := range nodeGrp2nodes {
+		if strings.HasPrefix(fullLabel, "kubernetes.io/hostname") {
```

Comment:
This will skip both `kubernetes.io/hostname` and `kubernetes.io/hostname-b` or `kubernetes.io/hostnames`, isnt it?

Reply:
Changed it to be an exact match.